pyshelf/search/formatter.py

```diff
@@ -0,0 +1,104 @@
+from distutils.version import LooseVersion
+from pyshelf.search.sort_type import SortType
+from pyshelf.search.sort_flag import SortFlag
+from pyshelf.search.type import Type as SearchType
+
+
+class Formatter(object):
+    def __init__(self, criteria, search_results, key_list=None):
+        """
+            Formats search results from Elasticsearch based on the given criteria.
+
+            Args:
+                criteria(dict): This contains the search and sort criteria as outlined on
+                                pyshelf.search.manager.Manager.search.
+                search_results(List[elasticsearch_dsl.result.Result]): List of search results.
+                key_list(list): list of keys to include in filtered results if list is not passed
+                                all fields will be returned.
+        """
+        self.search_criteria = criteria.get("search")
+        # Sort criteria must be reversed to ensure first sort criteria takes precedence
+        self.sort_criteria = reversed(criteria.get("sort", []))
+        self.key_list = key_list
+        self.search_results = search_results
+        self._results = None
+        self.tilde_search = {}
+
+    @property
+    def results(self):
+        """
+            Filtered and sorted results.
+        """
+        self._results = self._sort_results(self._filter_results())
+        return self._results
+
+    def _filter_version_search(self, item_name, item_value):
+        """
+            This ensures when a version search is done that any results that are
+            less then the version that is passed in are dropped from the result set.
+
+            Args:
+                item_name
```

Comment:
Looks like I started the doc block and just stopped. bdb3e3948a5c7eea380774407a99c1d59c6d3dfb